unity/Assets/Scripts/AgentManager.cs

```diff
@@ -568,6 +568,34 @@ Vector3 defaultsOnNull
         );
     }
 
+    //same as `GetCoordinateFromRaycast` for the main agent camera, but used with any ThirdPartyCamera in scene
+    public void CoordinateFromRaycastThirdPartyCamera(float x, float y, int thirdPartyCameraId = 0) {
+        // count is out of bounds
+        if (thirdPartyCameraId >= thirdPartyCameras.Count || thirdPartyCameraId < 0) {
+            throw new ArgumentOutOfRangeException(
+                $"thirdPartyCameraId: {thirdPartyCameraId} (int: default=0) must in 0 <= thirdPartyCameraId < len(thirdPartyCameras)={thirdPartyCameras.Count}."
+            );
+        }
+
+        if (x < 0 || y < 0 || x > 1 || y > 1) {
+            throw new ArgumentOutOfRangeException($"x and y must be in [0:1] not (x={x}, y={y}).");
+        }
+
+        Camera thirdPartyCamera = thirdPartyCameras[thirdPartyCameraId];
+
+        Ray ray = thirdPartyCamera.ViewportPointToRay(new Vector3(x, 1 - y, 0));
+        RaycastHit hit;
+        Physics.Raycast(
+            ray: ray,
+            hitInfo: out hit,
+            maxDistance: Mathf.Infinity,
+            layerMask: LayerMask.GetMask("Default", "Agent", "SimObjVisible", "PlaceableSurface"),
+            queryTriggerInteraction: QueryTriggerInteraction.Ignore
+        );
+
+        this.activeAgent().actionFinished(success: true, actionReturn: hit.point);
```

Comment:
Remind me: what happens if the raycast doesn't hit anything? Is `hit.point` a bunch of infinities?

Reply:
if it doesn't hit anything, the actual `Physics.Raycast` will return false, and the `RaycastHit hit` won't be updated, so `hit.point` will be at its default (0, 0, 0).

Reply:
In that case can we maybe return `null` rather than `hit.point`?

Reply:
yeah I think this would be safer, I'll add this in